Expose APIs

b1llow opened the pull request:
<!-- Filling this template is mandatory -->

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

<!-- Explain the **details** for making this change. Is a new feature implemented? What existing problem does the pull request solve? How does the pull request solve these issues? Please provide enough information so that others can review your pull request. -->

...

**Test plan**

<!-- What steps should the reviewer take to test your pull request? Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots/videos. This is your time to re-check that everything works and that you covered all the edge cases -->

...

**Closing issues**

<!-- put "closes #XXXX" in your comment to auto-close the issue that your PR fixes (if any). -->

...

b1llow committed:
expose `rz_core_bin_get_compile_time`

b1llow committed:
expose `rz_core_bin_get_compile_time`

b1llow committed:
expose APIs
- `rz_core_analysis_coverage_count`
- `rz_core_analysis_code_count`
- `rz_core_analysis_calls_count`

b1llow committed:
expose APIs
- `rz_core_analysis_coverage_count`
- `rz_core_analysis_code_count`
- `rz_core_analysis_calls_count`

b1llow committed:
expose API:
- `rz_core_sym_name_init`
- `rz_core_sym_name_fini`
- `rz_core_sym_is_export`

b1llow committed:
expose API:
- `rz_core_sym_name_init`
- `rz_core_sym_name_fini`
- `rz_core_sym_is_export`

b1llow committed:
expose `rz_core_bin_whole_strings`

b1llow committed:
expose `rz_core_bin_whole_strings`

b1llow committed:
Merge branch 'convert_apis2' of https://github.com/rizinorg/rizin into convert_apis2

b1llow committed:
expose `rz_core_config_variable_spaces`

b1llow committed:
doxygen

b1llow committed:
expose `rz_core_analysis_esil_deinit` and `rz_core_analysis_esil_init_mem_del`

b1llow committed:
expose `rz_core_types_as_c`

b1llow committed:
fix

b1llow committed:
fix

b1llow committed:
expose `rz_type_list_c_nl_handler`

b1llow committed:
expose `rz_core_file_open_load`

b1llow committed:
fix

ret2libc commented on librz/include/rz_core.h:
```diff
@@ -418,6 +418,20 @@ RZ_API void rz_core_item_free(RzCoreItem *ci);
+	char *classflag; // flag for classname
+	char *methname; // methods [class]::[method]
+	char *methflag; // methods flag sym.[class].[method]
+} SymName;
```
This should have a doxygen description, moreover any exported structure should start with `Rz`.

ret2libc commented on librz/include/rz_core.h:
```diff
@@ -457,6 +471,7 @@ RZ_API int rz_core_config_init(RzCore *core);
 RZ_API bool rz_core_config_eval_and_print(RzCore *core, const char *str, bool many);
 RZ_API void rz_core_config_print_all(RzConfig *cfg, const char *str, RzCmdStateOutput *state);
 RZ_API void rz_core_parse_rizinrc(RzCore *r);
+RZ_API RZ_OWN RzList *rz_core_config_variable_spaces(RZ_NONNULL RzCore *core, RZ_NULLABLE const char *space);
```
Shall this be named `rz_core_config_in_space`?

b1llow replied:
I don't know how to come up with a proper name

wargio replied:
i like ret2libc proposal.

b1llow committed:
refactor
`SymName` -> `RzSymName`
`rz_core_config_variable_spaces` -> `rz_core_config_in_space`

ret2libc commented on librz/include/rz_core.h:
```diff
@@ -418,19 +418,24 @@ RZ_API void rz_core_item_free(RzCoreItem *ci);
+	char *classflag; ///< flag for classname
+	char *methname; ///< methods [class]::[method]
+	char *methflag; ///< methods flag sym.[class].[method]
+} RzSymName;
```
@XVilka ? WDYT? Better name maybe? It has a very specific use case but the name is very generic, I'm not 100% sure it is good as-is.

XVilka replied:
I think, `RzBinSymNames` would be a better choice, note the *Names*

ret2libc replied:
RzCoreSymNames. It's in RzCore, not RzBin.

XVilka commented on librz/core/cbin.c:
```diff
@@ -1659,10 +1664,8 @@ RZ_API RZ_OWN HtPP *rz_core_bin_create_digests(RzCore *core, ut64 paddr, ut64 si
-	if (!core->bin) {
-		return false;
-	}
+RZ_API bool rz_core_bin_set_cur(RZ_NONNULL RzCore *core, RZ_NULLABLE RzBinFile *binfile) {
```
Missing Doxygen.

XVilka commented on librz/core/cbin.c:
```diff
@@ -1692,7 +1695,12 @@ static RZ_NULLABLE RZ_BORROW const RzList *core_bin_strings(RzCore *r, RzBinFile
 
-static const char *get_compile_time(Sdb *binFileSdb) {
+/**
+ * \brief Get TimeDateStamp string
```
I think this description is hard to understand for a newbie

XVilka commented on librz/core/cbin.c:
```diff
@@ -2854,13 +2862,20 @@ RZ_API bool rz_core_bin_whole_strings_print(RzCore *core, RzBinFile *bf, RzCmdSt
+RZ_API bool rz_core_bin_whole_strings_print(RzCore *core, RzBinFile *bf, RzCmdStateOutput *state) {
+	rz_return_val_if_fail(core && state, false);
+
+	RzList *l = rz_core_bin_whole_strings(core, bf);
```
It can return NULL, probably worth doing a quick return in this case

XVilka commented on librz/core/ctypes.c:
```diff
@@ -415,8 +415,14 @@ RZ_IPI RZ_OWN char *rz_core_base_type_as_c(RzCore *core, RZ_NONNULL RzBaseType *
+	rz_return_val_if_fail(core && core->analysis && name, NULL);
 
 	RzBaseType *btype = rz_type_db_get_base_type(core->analysis->typedb, name);
 	if (!btype) {
```
return NULL below, while you are on it.

XVilka reviewed (changes requested):
Looks great! Only a few small nitpicks and good to merge 👍

b1llow committed:
fix

b1llow committed:
fix

b1llow committed:
Merge branch 'dev' into convert_apis2

XVilka commented:
@imbillow please rebase to resolve conflicts, maybe squash too.

XVilka commented:
@imbillow once you do that and it passes CI - feel free to merge it yourself, just be sure to squash and have a good commit message.

XVilka committed:
Merge branch 'dev' into convert_apis2
